webpack/assets/javascripts/react_app/components/HostDetails/DetailsCard/index.js

```diff
@@ -120,18 +121,24 @@ const DetailsCard = ({
                     justifyContent={{ default: 'justifyContentSpaceBetween' }}
                   >
                     <FlexItem>
-                      <Button
-                        ouiaId="host-group-link"
-                        component="a"
-                        href={foremanUrl(
-                          `/hosts?search=hostgroup="${hostgroupName}"`
+                      <span>
+                        {hostgroupTitle?.substring(
+                          0,
+                          hostgroupTitle?.lastIndexOf(hostgroupName)
                         )}
-                        variant="link"
-                        target="_blank"
-                        isInline
-                      >
-                        {hostgroupName}
-                      </Button>
+                        <Button
+                          ouiaId="host-group-link"
+                          component="a"
+                          href={foremanUrl(
+                            `/hosts?search=hostgroup="${hostgroupName}"`
```

Comment:
There is still one issue with this. If you have a parent host group "parentA" and "parentB" its allowed to have a child host group with the name "child" in both "parentA" and "parentB" - so the child has the same name.
if you know have a button to "child" it will search for "child" having both, hosts of group "parentA/child" AND "parentB/child".

Reply:
Added projects issue: https://projects.theforeman.org/issues/38004

Reply:
@sbernhard thank you for noticing, I'm looking into that

Reply:
any update on this?